src/charm.py

```diff
@@ -131,37 +132,24 @@ def __init__(self, *args):
         self._crd_resource_handler = None
         self._configmap_resource_handler = None
 
-        metrics_port = ServicePort(int(self._metrics_port), name="metrics-port")
+        metrics_port = ServicePort(int(METRICS_PORT), name="metrics-port")
         webhook_port = ServicePort(int(self._webhook_port), name="webhook-port")
         self.service_patcher = KubernetesServicePatch(
             self,
             [metrics_port, webhook_port],
             service_name=f"{self.model.app.name}",
         )
 
-        # setup events to be handled by main event handler
-        self.framework.observe(self.on.config_changed, self._on_event)
-        for rel in self.model.relations.keys():
-            self.framework.observe(self.on[rel].relation_changed, self._on_event)
-
-        # setup events to be handled by specific event handlers
-        self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.upgrade_charm, self._on_upgrade)
-        self.framework.observe(self.on.seldon_core_pebble_ready, self._on_pebble_ready)
-        self.framework.observe(self.on.remove, self._on_remove)
-        self.framework.observe(self.on.stop, self._on_stop)
-
         # Prometheus related config
         self.prometheus_provider = MetricsEndpointProvider(
             charm=self,
             relation_name="metrics-endpoint",
             jobs=[
                 {
-                    "metrics_path": self.config["executor-server-metrics-port-name"],
-                    "static_configs": [{"targets": ["*:{}".format(self.config["metrics-port"])]}],
+                    "metrics_path": METRICS_PATH,
```

Comment:
This relates to the above question of mine regarding configurability of the metrics path and port, but I 'm wondering if using `self.config["executor-server-metrics-port-name"]` didn't work, since this had a [default value](https://github.com/canonical/seldon-core-operator/blob/29d7d3aa5a53df5ad404b713effbb640e094bda2/config.yaml#L103-L106) of `metrics`. I guess we would need to also add a `/` before that?!

Reply:
> but I 'm wondering if using self.config["executor-server-metrics-port-name"] didn't work,

It would've worked if it was correctly formatted, but also I decided to go with a var because the configuration option was very misleading `executor-server-metrics-port-name` is definitely not exactly the generic metrics endpoint of seldon. If we'd want to use a config option, we could've just gone with the `metrics-port` one.

> I guess we would need to also add a / before that?!

Yeah.